### ISSUE TITLE
Kind signal not recognized

### ISSUE THREAD

**Issue by Wolf2789** [matusnovak/doxybook2#15]
Hi, first of all I'd like to thank you for creating such a wonderful tool! :smile:
 
Recently I tried generating documentation for Qt-based application.
I've successfully generated XML files using Doxygen, but I'm getting this error when it comes to Doxybook2 part:
```
[...]
Loading classPressableLabel.xml
Warning at Doxygen.cpp:72 Failed to parse member classPressableLabel error: Exception at Enums.cpp:28 Kind signal not recognised please contact the author
Loading classMainWindow.xml
Warning at Doxygen.cpp:72 Failed to parse member classMainWindow error: Exception at Enums.cpp:28 Kind signal not recognised please contact the author
[...]
Finalizing...
Exception at Enums.cpp:269 Type 0 not recognised please contant the author!
```

Problematic XMLs:
* classPressableLabel.xml
```xml
[...]
      <sectiondef kind="signal">
      <memberdef kind="signal" id="classPressableLabel_1a70e5724a77a27c81cd8e324b427dfe90" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
        <type>void</type>
        <definition>void PressableLabel::pressed</definition>
        <argsstring>()</argsstring>
        <name>pressed</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription> <para>image has been pressed on with mouse </para> </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="pressablelabel.h" line="17" column="1"/>
      </memberdef>
      </sectiondef>
[...]
```
* classMainWindow.xml
```xml
[...]
      <sectiondef kind="signal">
      <memberdef kind="signal" id="classMainWindow_1a65a43bbc159bba7abbbdb54cdc9b68fa" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
        <type>void</type>
        <definition>void MainWindow::resized</definition>
        <argsstring>(QSizeF const &amp;)</argsstring>
        <name>resized</name>
        <param>
          <type>QSizeF const &amp;</type>
        </param>
        <briefdescription>
        </briefdescription>
        <detaileddescription> <para>signal emited whenever the window size has been changed</para><para><parameterlist kind="param"><parameteritem><parameternamelist><parametername>size</parametername></parameternamelist><parameterdescription><para>of the internal layout  to work properly, must be called with layout size </para></parameterdescription></parameteritem></parameterlist></para>        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="mainwindow.h" line="47" column="1"/>
      </memberdef>
      <memberdef kind="signal" id="classMainWindow_1af85fb6e3c6bcb51314d93c2c9b394ae6" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
        <type>void</type>
        <definition>void MainWindow::prevButtonPressed</definition>
        <argsstring>()</argsstring>
        <name>prevButtonPressed</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription> <para>signal emited when &apos;previous&apos; button is pressed </para>        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="mainwindow.h" line="50" column="1"/>
      </memberdef>
      <memberdef kind="signal" id="classMainWindow_1ab73b6a3371823c36713e8dea07ac85b7" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
        <type>void</type>
        <definition>void MainWindow::nextButtonPressed</definition>
        <argsstring>()</argsstring>
        <name>nextButtonPressed</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription> <para>signal emited when &apos;next&apos; button is pressed </para>        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="mainwindow.h" line="52" column="1"/>
      </memberdef>
      </sectiondef>
[...]
```

---

I believe the same applies to _slots_ kind.

```xml
[...]
      <sectiondef kind="public-slot">
      <memberdef kind="slot" id="classMainWindow_1a20e17289a70704d7f2afefb65c9d889c" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
        <type>void</type>
        <definition>void MainWindow::toggleFullscreen</definition>
        <argsstring>()</argsstring>
        <name>toggleFullscreen</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription> <para>toggle window fullscreen flag </para>        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="mainwindow.h" line="56" column="1" bodyfile="mainwindow.cpp" bodystart="119" bodyend="122"/>
      </memberdef>
      </sectiondef>
[...]
```

---

Also as a side note - you have few small typos in _Enums.cpp_ and _Generator.cpp_:
"[...] _please conta**n**t the author_ [...]" -> "[...] _please conta**c**t the author_ [...]"

Hope you'll be able to work this out :smiley:

**Comment by matusnovak** [comment]
Hi @Wolf2789 

Huh, I didn't know there can be `signal` and `slot` kinds in Doxygen. Are they only functions or can they by other types too? For example a class/struct field. If I am reading the [documentation](https://doc.qt.io/qt-5/signalsandslots.html) correctly, they can be only functions. Can you confirm that? I don't know how Qt works. 

If they are only functions, that will be very easy to add into Doxybook2. 

How would you want them to render in the markdown? As a separate section, such as `## Public Slots` and `## Public Signals` similar to how `## Public Functions` are rendered? ([Example](https://matusnovak.github.io/doxybook2/hugo-learn/classes/classengine_1_1audio_1_1audiobuffer/#public-functions-documentation)) Or would public/private/protected make no sense for signals and slots? I am asking because Doxygen's XML has `prot="public"` for signals and slots, which can be used.

**Comment by Wolf2789** [comment]
Thank you for the quick response.

It is exactly as you think. They are just like **public functions**, but also slots can be **virtual**.
For the rendering part - you're reading my mind 😃 And the example you've provided fits perfectly.
It would be nice to have separate sections for signals and for slots.

Thank you again, cheers 😄

**Comment by matusnovak** [comment]
Got it. I am going to implement it most likely tomorrow. I will also provide a branch and a binary release for you to test it out first, before I merge it to master.

**Comment by matusnovak** [comment]
Hi @Wolf2789 

I have added the support for signals and slots. It's in the following branch: [feature-signals-slots](https://github.com/matusnovak/doxybook2/tree/feature-signals-slots)

I have also made a binary release for you. (Sorry, no armv8 or OSX, can't compile for them at the moment).

* https://matusnovak-artifacts.s3-eu-west-1.amazonaws.com/doxybook2/doxybook2-linux-amd64-v1.1.5-3-gc1fdec7.zip
* https://matusnovak-artifacts.s3-eu-west-1.amazonaws.com/doxybook2/doxybook2-windows-win64-v1.1.5-3-gc1fdec7.zip

Let me know what you think.

Couple of questions:

* Does it make sense to have "Public Slots" and "Protected Slots"? Or are slots and signals only public? 
* In the original HTML output of Doxygen, the slots and signals do not have dedicated sections. They are all together with public functions in "Member Function Documentation" section (not the table that is at the top of the page, but the detailed documentation at the bottom). In the changes I have made, Doxybook2 is creating a dedicated section for them. Should they belong to their own sections or be together in one single section with other member functions?

**Comment by Wolf2789** [comment]
Hi, sorry for the late reply.

Answering your questions:
1. Signals and slots are always public.
2. For me, the way you're generating them right now is just perfect. In fact they are just functions, but they have special purpose, so having them in separate section makes everything more clear and readable.

As for the release you've provided: everything works perfectly, but I'm curious, why in every release, you're putting Doxygen include directory under another include directory? Shouldn't it be just **include/Doxygen** instead of **include/include/Doxygen**?

**Comment by matusnovak** [comment]
Awesome. Thanks for testing it. I will merge it to master and make a release. 

> you're putting Doxygen include directory under another include directory? Shouldn't it be just include/Doxygen instead of include/include/Doxygen?

Yeah... my bad... I did not notice that.

**Comment by matusnovak** [comment]
@Wolf2789 

I have merged it and released it as [releases/tag/v1.1.6](https://github.com/matusnovak/doxybook2/releases/tag/v1.1.6)
